src/lib.rs

```diff
@@ -1,3 +1,130 @@
+    fn get_readings(&self) -> Result<TypedReadingsResult<f64>, SensorError> {
+        let mut temperature: f32 = 0.0;
+        let mut pressure: f32 = 0.0;
+        let mut humidity: f32 = 0.0;
+        unsafe {
+            log::debug!("bme280 - reading temperature...");
+            esp!(bme280_read_temperature(
+                BME280,
+                &mut temperature as &mut f32
+            ))?;
+            log::debug!("bme280 - reading pressure...");
+            esp!(bme280_read_pressure(BME280, &mut pressure as &mut f32))?;
+            log::debug!("bme280 - reading humidity...");
+            esp!(bme280_read_humidity(BME280, &mut humidity as &mut f32))?;
+        }
+        log::debug!(
+            "temperature: {}, humidity: {}, pressure: {}",
+            temperature,
+            humidity,
+            pressure
+        );
+        let mut x: HashMap<String, f64> = HashMap::new();
+        x.insert("temperature_c".to_string(), temperature.into());
+        x.insert("humidity".to_string(), humidity.into());
+        x.insert("pressure_hpa".to_string(), pressure.into());
+        Ok(x)
+    }
+}
+
+impl Bme280 {
+    pub fn from_config(cfg: ConfigType, _deps: Vec<Dependency>) -> Result<SensorType, SensorError> {
+        // DO NOT use the board i2c interface to initialize the i2c bus, the idf-component will handle it
+        let bus_no = cfg
+            .get_attribute::<i32>("i2c_bus")
+            .inspect_err(|_| {
+                log::warn!("`i2c_bus` attribute not found or invalid, defaulting to bus 0")
+            })
+            .unwrap_or_default();
+
+        let sda_pin = cfg
+            .get_attribute::<i32>("sda_pin")
+            .inspect_err(|_| {
+                log::info!("`sda_pin` attribute not found or invalid, defaulting to pin 22")
+            })
+            .unwrap_or(22);
+        let scl_pin = cfg
+            .get_attribute::<i32>("scl_pin")
+            .inspect_err(|_| {
+                log::info!("`scl_pin` attribute not found or invalid, defaulting to pin 21")
+            })
+            .unwrap_or(21);
+
+        let config = i2c_config_t {
+            mode: I2C_MODE_MASTER,
+            sda_io_num: sda_pin,
+            sda_pullup_en: true,
+            scl_io_num: scl_pin,
+            scl_pullup_en: true,
+            __bindgen_anon_1: i2c_config_t__bindgen_ty_1 {
+                master: i2c_config_t__bindgen_ty_1__bindgen_ty_1 { clk_speed: 100000 },
+            },
+            clk_flags: 0,
+        };
+
+        unsafe {
+            log::info!("initializing I2C_BUS: {:?}", I2C_BUS);
+            I2C_BUS = i2c_bus_create(bus_no, &config);
+            log::info!("initialized I2C_BUS to : {:?}", I2C_BUS);
+            log::info!("initializing BME280: {:?}", BME280);
+            BME280 = bme280_create(I2C_BUS, BME280_I2C_ADDRESS_DEFAULT.try_into().unwrap());
+            log::info!("initialized BME280 to : {:?}", BME280);
+            esp!(bme280_default_init(BME280))?;
```

Comment:
nit: you have before/after log statements for everything but this init step, for consistency log here too